frontend/src/metabase/visualizations/echarts/cartesian/utils/timeseries.ts

```diff
@@ -197,31 +162,40 @@ function timeseriesTicksInterval(
     initialIndex = 0;
   }
 
-  // now starting at the TIMESERIES_INTERVALS entry in question, calculate the expected tick count for that interval
-  // based on the time range we are displaying. If the expected tick count is less than or equal to the target
-  // maxTickCount, we can go ahead and use this interval. Otherwise continue on to the next larger interval, for
-  // example every 3 hours instead of every one hour. Continue until we find something with an interval large enough
-  // to keep the total tick count under the max tick count
-  for (const interval of _.rest(TIMESERIES_INTERVALS, initialIndex)) {
-    if (expectedTickCount(interval, timeRangeMilliseconds) <= maxTickCount) {
-      return interval;
+  // Fallback value: the largest tick interval (every 100 years)
+  let intervalIndex = TIMESERIES_INTERVALS.length - 1;
+
+  // Looking for the first interval which produces less ticks than the maxTicksCount.
+  // However, if it produces less than minTickCount ticks we prefer taking a smaller previous interval.
+  for (let i = initialIndex; i < TIMESERIES_INTERVALS.length; i++) {
+    const intervalTicksCount = expectedTickCount(
+      TIMESERIES_INTERVALS[i],
+      timeRangeMilliseconds,
+    );
+
+    if (intervalTicksCount > maxTickCount) {
+      continue;
+    }
+
+    if (intervalTicksCount <= maxTickCount) {
```

Comment:
Nit: I think it would be easier to read if these were two separate if statements, first `intervalTicksCount === maxTickCount` then `intervalTicksCount < maxTickCount`, then we can avoid the creating the `isTooSpare` boolean and ternary operator

Reply:
It will be slightly different, here we find the first interval which produces no more than `**max**TickCount` and then we check whether the count of ticks is too sparse by comparing it to **min** ticks count allowed `intervalTicksCount < **min**TickCount`. If it is too sparse (<2 ticks per chart) we select the previous interval that produced more than needed ticks which is ok because ECharts `hideOverlap` hides overlapping ones